databallpy/load_data/tracking_data/_quality_check_tracking_data.py

```diff
@@ -0,0 +1,184 @@
+    valid_frames = ~np.isnan(
+        tracking_data.loc[mask_ball_alive, ["ball_x", "ball_y"]]
+    ).any(axis=1)
+    sum_valid_frames = sum(valid_frames)
+    n_total_frames = len(tracking_data.loc[mask_ball_alive])
+    if sum_valid_frames < n_total_frames * 0.99:
+        warnings.warn(
+            DataBallPyWarning(
+                "Ball data is not available for more than 1% of all frames"
+            )
+        )
+
+    max_nan_sequence = _max_sequence_invalid_frames(valid_frames)
+    if max_nan_sequence >= 5 * framerate:
+        warnings.warn(
+            DataBallPyWarning("There is a gap in the ball data for at least 5 seconds")
+        )
+
+    return
+
+
+def _check_ball_velocity(tracking_data: pd.DataFrame, framerate: float) -> None:
+    """Function that checks the ball data by checking it's velocity
+
+    Args:
+        tracking_data (pd.DataFrame): tracking data
+        framerate (float): framerate of the trackin data
+    """
+    initial_columns = tracking_data.columns
+    input_columns = ["ball_x", "ball_y"]
+    tracking_data = get_velocity(tracking_data, input_columns, framerate)
+    velocity_ball = np.hypot(tracking_data["ball_x_v"], tracking_data["ball_y_v"])
+    mask_ball_alive = tracking_data["ball_status"] == "alive"
+    valid_frames = velocity_ball[mask_ball_alive][1:] < 50
+    sum_valid_frames = sum(valid_frames)
+    n_total_frames = len(valid_frames)
+
+    if sum_valid_frames < n_total_frames * 0.99:
+        warnings.warn(
+            DataBallPyWarning(
+                "Ball velocity is unrealistic (> 50 m/s\N{SUPERSCRIPT TWO}) for \
+                    more than 1% of all frames"
+            )
+        )
+
+    max_invalid_sequence = _max_sequence_invalid_frames(valid_frames)
+
+    if max_invalid_sequence > 5 * framerate:
+        warnings.warn(
+            DataBallPyWarning(
+                "There is a gap in the realistic ball data \
+                    (velocity < 50 m/s\N{SUPERSCRIPT TWO}) for at least 5 seconds"
+            )
+        )
+
+    tracking_data = tracking_data[initial_columns]
+    return
+
+
+def _check_player_velocity(
+    tracking_data: pd.DataFrame, framerate: float, periods: pd.DataFrame
+) -> None:
+    """Function that checks all the player tracking data by checking the velocities
+
+    Args:
+        tracking_data (pd.DataFrame): tracking data
+        framerate (float): framerate of the tracking data
+        periods (pd.DataFrame): holds information on start and end of all periods
+    """
+    initial_columns = tracking_data.columns
+    player_columns = [x for x in tracking_data.columns if "home" in x or "away" in x]
+    players = [x.replace("_x", "") for x in player_columns if "_x" in x]
+    tracking_data = get_velocity(tracking_data, player_columns, framerate)
+
+    mask_no_break = [False] * len(tracking_data)
+    first_frame = periods.loc[0, "start_frame"]
+    for _, row in periods.iterrows():
+        if row["start_frame"] != -999:
```

Comment:
Maybe raise an warning is this is true, because now everything is considered break, therefore you have no frames to check the velocity of, so it did not really pass the sanity check right?